app/src/acsmatmult/utils/Matrix.h

```diff
@@ -20,41 +20,30 @@
 #include <random>
 #include <cstring>
 
-#include "RandomGenerator.hpp"
+#include "acsmatmult/utils/RandomGenerator.h"
 
 /**
  * @brief A matrix
  * @tparam T The type of the elements of the matrix, can be float or double
  */
 template<typename T>
 struct Matrix {
-  /// @brief Number of rows in this matrix
-  size_t rows = 0;
-
-  /// @brief Number of columns in this matrix
-  size_t columns = 0;
-
-  /// @brief Raw data buffer of this matrix
-  std::shared_ptr<T> values = nullptr;
+  size_t rows = 0;  ///< Number of rows in this matrix
+  size_t columns = 0;  ///< Number of columns in this matrix
+  std::shared_ptr<T> values = nullptr;  ///< Raw data buffer of this matrix
 
   /**
-   * Construct a new matrix
+   * @brief Construct a new matrix
    * @param rows The number of rows of the matrix
    * @param columns The number of columns of the matrix
    */
   Matrix<T>(size_t rows, size_t columns) : rows(rows), columns(columns) {
     // Make sure the dimensions are valid
-    if ((rows > 0) && (columns > 0)) {
-      // Allocate space, zero initialize it and store matrix using a smart pointer
-      values = std::shared_ptr<T>(new T[rows * columns](), std::default_delete<T[]>());
-
-      // Check if calloc worked.
-      if (values == nullptr) {
-        throw std::runtime_error("Unable to allocate memory for matrix.");
-      }
-    } else {
+    if ((rows <= 0) && (columns <= 0)) {
```

Comment:
```suggestion
    if ((rows <= 0) || (columns <= 0)) {
```
I suppose the error should also be thrown when just one of the dimensions is negative or zero.